### DISCUSSION TITLE
Embedded CUE Sheet Support

### DISCUSSION THREAD

**Discussion by G2-Games** [Serial-ATA/lofty-rs#285] [Ideas]
Unless I missed something, I did not see any mention of being able to read embedded CUE sheets using Lofty. Is this a feature you would consider adding?

I currently have no good way to test this feature, but I can provide a file that includes a sheet like that.

References:
[FLAC Spec](https://xiph.org/flac/format.html#metadata_block_cuesheet)

**Comment by Serial-ATA** [comment]
This seems simple enough to parse, at least.

Since this would be stored in `FlacFile`, two things would need to happen first, though:

1. We need some way to indicate whether or not the caller is using `read_from{_path}` or `Probe::read()` since the cue block is unnecessary to parse in those cases.

2. Since lofty only works with tags, the writing is done in their context (eg. `FlacFile::write_to` simply calls `VorbisComments::write_to(file)`). There'd have to be some changes to allow for a different, format-specific field to be written at the same time as the tag.
